crates/rpc/src/middleware/versioning.rs

```diff
@@ -60,7 +67,13 @@ pub(crate) async fn prefix_rpc_method_names_with_version(
             return Err(BoxError::from(VersioningError::HealthCheck));
         }
         // RPC endpoints
-        "/" | "/rpc/v0.3" | "/rpc/v0.3/" => &[("starknet_", "v0.3_"), ("pathfinder_", "v0.3_")][..],
+        "/" if default_version == DefaultVersion::V03 => {
```

Comment:
It's becoming messier and messier :fearful: I hope we move away from this code soon :crossed_fingers: .